core/src/main/scala/org/apache/spark/TestUtils.scala

```diff
@@ -418,17 +419,18 @@ private[spark] object TestUtils {
   }
 
   /**
-   * config a log4j properties used for testsuite
+   * config a log4j2 properties used for testsuite
    */
-  def configTestLog4j(level: String): Unit = {
-    val pro = new Properties()
-    pro.put("log4j.rootLogger", s"$level, console")
-    pro.put("log4j.appender.console", "org.apache.log4j.ConsoleAppender")
-    pro.put("log4j.appender.console.target", "System.err")
-    pro.put("log4j.appender.console.layout", "org.apache.log4j.PatternLayout")
-    pro.put("log4j.appender.console.layout.ConversionPattern",
-      "%d{yy/MM/dd HH:mm:ss} %p %c{1}: %m%n")
-    PropertyConfigurator.configure(pro)
+  def configTestLog4j2(level: String): Unit = {
+    val builder = ConfigurationBuilderFactory.newConfigurationBuilder()
+    val appenderBuilder = builder.newAppender("console", "CONSOLE")
+      .addAttribute("target", ConsoleAppender.Target.SYSTEM_ERR)
+    appenderBuilder.add(builder.newLayout("PatternLayout")
+      .addAttribute("pattern", "%d{yy/MM/dd HH:mm:ss} %p %c{1}: %m%n"))
+    builder.add(appenderBuilder)
+    builder.add(builder.newRootLogger(s"$level").add(builder.newAppenderRef("console")))
+    val configuration = builder.build()
+    LogManager.getContext(false).asInstanceOf[LoggerContext].reconfigure(configuration)
```

Comment:
This rewriting looks okay. I just wonder the purpose of `configTestLog4j` and if it is necessary. These tests are not checking the log (as they are still passed with no-op `configTestLog4j` now). And seems it is not used to reduce verbose logging.

Reply:
Maybe we can just remove `configTestLog4j`?

Reply:
I traced back to the original pr ([SPARK-3193](https://github.com/apache/spark/pull/2108)) it seems that `configTestLog4j ` is only added to help troubleshoot the failed UT on Jenkins.

I agree to remove this method directly, what do you think @dongjoon-hyun ?

Reply:
If we are sure to remove the `configTestLog4j `directly, I will update the pr and pr description

Reply:
Hmm... It seems that the method of calling `configTestLog4j` in the current UTs is in the process of `runSparkSubmit`.  If we directly remove this method, I can not find out where to see the relevant log of `runSparkSubmit` (not in `target/units`), which is really not conducive to troubleshooting.

Reply:
Then it is okay to keep it, I think.

Reply:
ok ~